### ISSUE TITLE
inconsistent ad state (sometimes midroll sometimes adsdone)

### ISSUE THREAD

**Issue by yanfaingold** [videojs/videojs-contrib-ads#452]
### Description

Ads state is wrong in the following vast if the click occurs after content ended (if not the state say Midroll, if ends it says AdsDone)

### Steps to Reproduce

the vast: https://pubads.g.doubleclick.net/gampad/ads?sz=640x480&iu=/124319096/external/single_ad_samples&ciu_szs=300x250&impl=s&gdfp_req=1&env=vp&output=vast&unviewed_position_start=1&cust_params=deployment%3Ddevsite%26sample_ct%3Dnonlinearvpaid2js&correlator=

just use the following code to see the stringed state: player.ads._state.constructor.name
in the debugger at any moment of the video.

#### Expected Results

the state is not consistent, if you click the ad mid content you will see the ad state is Midroll but if you click after the content ended the ad state is AdsDone.

#### Actual Results

The state should be the same.

**Comment by yanfaingold** [comment]
My full flow is including a usage of prebid.js + videojs + videojs-ima + contrib-ads but im cutting the flow for checks by directly passing an xml which contains a vast/vmap to the videojs-ima plugin to check certain vasts/vmaps.

**Comment by yanfaingold** [comment]
Yes i noticed its not part of the public API, but it can be useful for tracking the ads when the events are not enough, in my case there were discrepencies in the data, for example adding all the ad end and ad error events != to the ad rendered events count.

**Comment by incompl** [comment]
If the current public API is insufficient, let's figure out what we need to add. Do you know why ad ends + ad errors was insufficient?
